prow/plugins/transfer-issue/transfer-issue.go

```diff
@@ -0,0 +1,110 @@
+	GetRepo(org, name string) (github.FullRepo, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsCollaborator(org, repo, user string) (bool, error)
+	TransferIssue(org, dstRepoNodeID string, issueNodeID string) (*github.TransferIssueMutation, error)
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The transfer-issue plugin transfers a GitHub issue from one repo to another in the same organization.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/transfer[-issue] <destination repo in same org>",
+		Description: "Transfers an issue to a different repo in the same org.",
+		Featured:    true,
+		WhoCanUse:   "Org members.",
+		Examples:    []string{"/transfer-issue kubectl", "/transfer test-infra"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handleTransfer(pc.GitHubClient, pc.Logger, e)
+}
+
+func handleTransfer(gc githubClient, log *logrus.Entry, e github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	srcRepoName := e.Repo.Name
+	srcRepoPair := org + "/" + srcRepoName
+	user := e.User.Login
+
+	if e.IsPR || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	matches := transferRe.FindAllStringSubmatch(e.Body, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+	if len(matches) != 1 || len(matches[0]) != 2 {
+		log.Warnf("invalid usage: %v", matches)
+		return gc.CreateComment(
+			org, srcRepoName, e.Number,
+			plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, "/transfer-issue must only be used once and with a single destination repo."),
+		)
+	}
+
+	dstRepoName := matches[0][1]
+	dstRepoPair := org + "/" + dstRepoName
+
+	dstRepo, err := gc.GetRepo(org, dstRepoName)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch destination repo: %s", dstRepoPair)
+		// TODO: Might want to add another GetRepo type call that checks if a repo exists vs a bad request
+		return gc.CreateComment(
+			org, srcRepoName, e.Number,
+			plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, fmt.Sprintf("Something went wrong or the destination repo %s does not exist.", dstRepoPair)),
+		)
+	}
+
+	isCollaboratorSrc, err := gc.IsCollaborator(org, srcRepoName, user)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch if user: %s is collaborator of source repo: %s", user, srcRepoPair)
+		return err
+	}
+	isCollaboratorDst, err := gc.IsCollaborator(org, dstRepoName, user)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch if user: %s is collaborator of destination repo: %s", user, dstRepoPair)
+		return err
+	}
+	if !(isCollaboratorSrc && isCollaboratorDst) {
+		log.Warnf("user %s is not a collaborator of %s and/or %s", user, srcRepoPair, dstRepoPair)
```

Comment:
This doesn't warrant a warning or a log at all, as there is nothing an administrator of Prow can or should do about this